src/es-module-shims.js

```diff
@@ -97,6 +100,7 @@ function metaResolve (id, parentUrl = this.url) {
 
 importShim.resolve = resolveSync;
 importShim.getImportMap = () => JSON.parse(JSON.stringify(importMap));
+importShim.setImportMap = (importMapIn) => Object.assign(importMap, JSON.parse(JSON.stringify(importMapIn)));
```

Comment:
```suggestion
importShim.setImportMap = importMapIn => {
  if (!shimMode) throw new Error('Unsupported in polyfill mode.');
  importMap = JSON.parse(JSON.stringify(importMapIn));
}
```

Reply:
I used Object.assign in order to preserve the existing `importMap` object defined as `let importMap = { imports: {}, scopes: {} };` (in fact to preserve the reference to the object, in case someone was referencing it). But I think this assumption is futile. I will change as you suggested.